r-reproducibility.Rmd

```diff
@@ -0,0 +1,1211 @@
+
+We'll cover the first three items in this course.
+
+## Project organization
+
+First off, what exactly does "project" mean? That depends a bit on the group,
+individual, or situation, but for our purposes, a "project" is anything related
+to completed one or more scientific "products" (e.g. poster, slides, manuscript,
+package, teaching material) related to a specific question or goal. This could
+be "one manuscript publication and associated conference presentations" per project.
+Keeping a project confined to one "scientific output" ensures that the project
+is kept reproducible, all files will relate to that "output", and can be easily
+archived once the manuscript has been published. However, this definition could
+be different depending on your own situation and goals.
+
+TODO: refs to rrtools, prodigenr
+
+The ability to read, understand, modify, and write simple pieces of code is an
+essential skill for modern data analysis tasks and projects. Here we introduce
+you to some of the best practices one should have while writing their code.
+Many of these best practices were taken from published "best practices" articles
+[@Nobl2009;@Tasc2017;@Wils2017]
+
+TODO: Other references for best practices?
+
+- Organise all R scripts and files in a single parent directory using a common
+and consistent folder and file structure.
+- Use [version control][version-control] to track changes to files.
+- Make raw data "read-only" (don't edit it directly) and use code to show what
+was done.
+- Write and describe code for people to read by being descriptive and using a
+[style guide][r-style-guide].
+- Think of code as part of your research product: Write for an audience 
+or other reader.
+- Don't repeat yourself by using and creating functions.
+- Whenever possible, use code to create output (figures, tables) rather than
+manualling creating or editing them.
+
+Managing your projects in a reproducible fashion doesn't just make your science
+reproducible, it also makes your life easier! RStudio is here to help us with
+that by using [R Projects][rstudio-r-projects]. RStudio projects make it
+straightforward to divide your work into multiple contexts, each with their own
+working directory, workspace, history, and source documents.
+
+It is strongly recommended that you store *all* the necessary files that will be
+used in your code in the **same parent directory**. You can then use
+relative file paths to access them (we'll talk about file paths below). This
+makes the directory and R Project a "product" or "bundle/package". Like a tiny
+machine, that needs to have all its parts in the same place.
+
+### Creating your first project
+
+There are many ways one could organise a project folder. We'll set up a project
+directory folder using the
+[prodigenr](http://prodigenr.lukewjohnston.com) package:
+
+```r
+# prodigenr::setup_project("ProjectName")
+prodigenr::setup_project("learningr")
+```
+
+TODO: Use prodigenr or something else?
```

Comment:
Put three options on a Twitter poll and see what people prefer?

Reply:
"three options"... the reason I made prodigenr is because everyone (that I've encountered) does it all differently because no one is taught how to do it a better or more standard way. And even among people that are more advanced in this, they all have very different approaches. The next alternative would be `usethis::create_project()` (which `prodigenr::set_project()` also uses among other things under the hood) but it makes an even simpler layout. The todo was more a note to myself to think about this more.